scripts/init-submodules-no-riscv-tools.sh

```diff
@@ -59,13 +40,7 @@ git config --unset submodule.generators/sha3.update
 # Shallow clone by default (recursive clone would get linux)
 git submodule update --init generators/sha3
 
-if [ $NO_FIRESIM = false ]; then
-  echo "initializing firesim"
-  # Renable firesim and init only the required submodules to provide
-  # all required scala deps, without doing a full build-setup
-  git config --unset submodule.sims/firesim.update
-  git submodule update --init sims/firesim
-  git -C sims/firesim submodule update --init sim/midas
-  git -C sims/firesim submodule update --init --recursive sw/firesim-software
-  git config submodule.sims/firesim.update none
-fi
+git config --unset submodule.sims/firesim.update
+# Non-recursive clone for minimal dependencies (midas)
+git submodule update --init sims/firesim
+git -C sims/firesim submodule update --init sw/firesim-software
```

Comment:
Per offline discussion with @NathanTP, the upcoming release will still clone FireMarshal by default to avoid breaking user expectations.

When FireMarshal migrates to Chipyard in the subsequent release, it would make sense to initialize FireMarshal along with the toolchain build, perhaps renaming `build-toolchains.sh` to `setup-sw.sh` and this script to `setup-hw.sh`.